### PR TITLE
Chem rework mk 4 - Cry of the larba

### PR DESCRIPTION
**Pull request by jroinc**
## About The Pull Request

Alters marine chem access and reagants to lock roundstart access to some stuff behind medics.

## Why It's Good For The Game

It's currently possible for marines to make dozens of injectors prior to opening the shutters that heal 3.8 brute/burn/tick with tramadol. This appears to let marines outheal xenos in most situations, while on the move. Additionally, it's possible to hit 5.3/tick via pills This will remove access to that heal rate roundstart and softcap non-medic healing injectors to 1.8 brute/burn/tick (2.3 with pills). In order to access the higher-tier chems, get larba corpses and wring the delicious fluid out of their tiny little bodies.

## Changelog
:cl:

balance: rebalanced chemistry

/:cl:

Larval soup reagant composition changed.
Larva have a T1 raw resource, and a T2 processed resource.
Drinking larval jelly is bad for you. If you're infected, it's SUUPER bad for you and you will die rapidly.
Drinking prepared larval jelly heals you slightly. Yay!

Altered the recipes for Meralyne and Dermaline to require basic larval jelly to minimize the whole "25 unga injectors before drop" meme.

Added a recipie for neuraline for if you have like 20 larba and need a boost planetside to finish them off. Neuraline healing and toxin damage effects also upped  by 50%.

Oxycodone made less bad. Stamina loss halved, should be usable in injector quantities now. Ideally, need to replace with an accuracy and scatter penalty later, once I figure out how to do that.

Kelotane and Dermaline temperature adjust halved. It was possible to become near-completely immune to fire while on both for minutes at a time. Use leporazine for that.

Meralyne and Dermaline heal rate slightly improved due to increased rarity. Constant at 2/tick when below 10u. Second-tier effect threshold lowered to 10u from 15u, and heal-rate adjusted up by 0.5/tick for a net change of +0.5 above 10u. Microdose injectors will perform the same, but single-dose pills (or 2-dose injectors) can now take advantage of the T2 healrates.

Antitox typepaths changed to purge the deadly stuff. I really need to fix david's shoddy fix of my hacky coffee implementation, but this is an acceptable net for now.

Synap description changed, stamina gains and losses upped by 50%.

Perdiox+ altered. Toxin amount from normal use removed, but amount added to OD damage. Instead, normal usage generates 5 toxin (the chem) per tick (total of 50). Take antitox with it and there isn't a downside. Forget antitox and you'll have a bad day. Small amount of arithrazane added to the injector by default to give a bit of a buffer for about a minute.

Hyperzine movespeed amount glared at angrily, re-work considered for later.

Ultrazine movespeed amount halved.

Polyhexanide error fixed.

Leporazine OD made not cancer, stun dropped from 30 seconds to 5.

Removed Advanced Injector pouch from purchasable medic gear. Added it as a free item to replace his autoinjector pouches from clothing vendor.

Upped the price on cases of pure chems, to lower the chances of people just vending them then making 20 injectors anyway. Still doable, but it'll be most of your loadout.
Neuraline slightly cheapened as a consequence of the above change.

~~Hypervene purge rate, chem purge rate, and inject/pill amounts rebalanced. One shot will purge 60u of chems, or 20u of larval toxin. 2 shots taken at the same time will OD you and almost certainly make you vomit, stunning you. Should be much easier to remove Larval now.~~ It's partially broken. Will fix later.

Replaced advanced injector pouch's hyperzine injector with perdiox+ injector. Can't wait to see how that all goes wrong~

Adds experimental buff to Paracetamol. It purges T1 healing meds at 1u/tick, but now heals slightly and restores 1 stamina/tick as the only long-lasting stamina restoration drug.

### CODE REVIEW

File: code/game/objects/machinery/vending/new_marine_vendors.dm [hunk 1]
```diff
@@ -791,17 +791,16 @@ GLOBAL_LIST_INIT(marine_selector_cats, list(
 
 		/obj/item/storage/firstaid/regular = list(CAT_MEDSUP, "Firstaid kit", 2, "black"),
 		/obj/item/storage/firstaid/adv = list(CAT_MEDSUP, "Advanced firstaid kit", 4, "orange"),
-		/obj/item/storage/pouch/autoinjector/advanced/full = list(CAT_MEDSUP, "Advanced Medical Injectors", 15, "orange"),
 		/obj/item/storage/pill_bottle/paracetamol = list(CAT_MEDSUP, "Paracetamol pills", 10, "black"),
-		/obj/item/storage/syringe_case/meralyne = list(CAT_MEDSUP, "syringe Case (120u Meralyne)", 9, "orange"),
-		/obj/item/storage/syringe_case/dermaline = list(CAT_MEDSUP, "syringe Case (120u Dermaline)", 9, "orange"),
-		/obj/item/storage/syringe_case/meraderm = list(CAT_MEDSUP, "syringe Case (120u Meraderm)", 9, "orange"),
+		/obj/item/storage/syringe_case/meralyne = list(CAT_MEDSUP, "syringe Case (120u Meralyne)", 16, "orange"),
```
Comment (Pariah919) [inline code/game/objects/machinery/vending/new_marine_vendors.dm:795]: Undocumented changes on this file.

File: code/modules/reagents/reagents/medical.dm [hunk 2]
```diff
@@ -288,7 +288,7 @@
 	description = "Dylovene is a broad-spectrum antitoxin."
 	color = "#A8F59C"
 	scannable = TRUE
-	purge_list = list(/datum/reagent/toxin, /datum/reagent/toxin/xeno_neurotoxin, /datum/reagent/consumable/drink/coffee)
+	purge_list = list(/datum/reagent/toxin, /datum/reagent/toxin/xeno_neurotoxin, /datum/reagent/consumable/drink/atomiccoffee, /datum/reagent/consumable/larvajelly)
```
Comment (Pariah919) [inline code/modules/reagents/reagents/medical.dm:291]: Undocumented changes.

File: code/modules/reagents/reagents/medical.dm [hunk 3]
```diff
@@ -565,7 +565,7 @@
 	scannable = TRUE
 
 /datum/reagent/medicine/peridaxon_plus/on_mob_life(mob/living/L, metabolism)
-	L.adjustToxLoss(20*REM)
+	L.reagents.add_reagent(/datum/reagent/toxin,5)
```
Comment (Pariah919) [inline code/modules/reagents/reagents/medical.dm:568]: Again. Undocumented.

**Comment by Pariah919** [comment]
The changelog is broken and this PR has a ton of undocumented changes.

File: code/game/objects/machinery/vending/new_marine_vendors.dm (hunk 1 above)
Comment (jroinc) [reply code/game/objects/machinery/vending/new_marine_vendors.dm:795]: Here's the documentation~

Upped the price on cases of pure chems, to lower the chances of people just vending them then making 20 injectors anyway. Still doable, but it'll be most of your loadout.

File: code/modules/reagents/reagents/medical.dm (hunk 2 above)
Comment (jroinc) [reply code/modules/reagents/reagents/medical.dm:291]: Here's the documentation~

Upped the price on cases of pure chems, to lower the chances of people just vending them then making 20 injectors anyway. Still doable, but it'll be most of your loadout.

File: code/modules/reagents/reagents/medical.dm (hunk 3 above)
Comment (jroinc) [reply code/modules/reagents/reagents/medical.dm:568]: Perdiox+ altered. Toxin amount from normal use removed, but amount added to OD damage. Instead, normal usage generates 5 toxin (the chem) per tick (total of 50). Take antitox with it and there isn't a downside. Forget antitox and you'll have a bad day.

**Comment by jroinc** [comment]
> The changelog is broken and this PR has a ton of undocumented changes.

Everything you listed is actually documented in the description.

**Comment by jroinc** [comment]
> The changelog is incredibly undescriptive and this PR has a ton of undocumented changes.
> Medic vendor syringe cases have been nerfed to 16. This isn't documented is the main one.
> Neuraline is less.

Upped the price on cases of pure chems, to lower the chances of people just vending them then making 20 injectors anyway. Still doable, but it'll be most of your loadout.
Neuraline slightly cheapened as a consequence of the above change.

If you don't feel like reading, I can't... I can't help you.

**Comment by Pariah919** [comment]
>Upped the price on cases of pure chems, to lower the chances of people just vending them then making 20 injectors anyway. Still doable, but it'll be most of your loadout.
Neuraline slightly cheapened as a consequence of the above change.

>If you don't feel like reading, I can't... I can't help you.
shrug. State the actual changes instead of paraphrasing. Don't care.

**Comment by factoryman942** [comment]
🤔

**Comment by Hathkar** [comment]
Larva jelly is coded to act as a catalyst for derma/mera/neura right now, meaning one dose can make infinite meds.

**Comment by jroinc** [comment]
> Larva jelly is coded to act as a catalyst for derma/mera/neura right now, meaning one dose can make infinite meds.

Yeah, I changed it to that. Having it be a reagent was too punishing for MD's to work out, and it still serves it's purpose of locking chems to ayy contact.

**Pull request merged**